Assets/_Project/Scripts/Player/FPMovementState.cs

```diff
@@ -0,0 +1,98 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public abstract class FPMovementState
+{
+    protected FPMovementContext context;
+    public FPMovementContext Context { get { return context; } set { context = value; } }
+
+    public abstract void Enter();
+    public abstract void Update();
+    public abstract void PhysicsUpdate();
+    public abstract void Exit();
```

Comment:
Since most of the states don't actually implement these 4 functions they should all be `virtual` with empty bodies instead of `abstract`, then you won't have to `override` them in the subclasses that don't use them